batch7crud-roll9/src/main/java/com/lftechnology/batch7crud/util/StudentValidator.java

```diff
@@ -0,0 +1,77 @@
+package com.lftechnology.batch7crud.util;
+
+import com.lftechnology.batch7crud.constants.AppConstant;
+import com.lftechnology.batch7crud.constants.AttributeConstant;
+import com.lftechnology.batch7crud.constants.MessageConstant;
+import com.lftechnology.batch7crud.constants.ParameterConstant;
+import com.lftechnology.batch7crud.entity.Student;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by sanjay on 1/22/16.
+ */
+public class StudentValidator {
+  public Student createObject(HashMap error, HashMap input) {
+    Student student = new Student();
+    Set set = input.entrySet();
+    Iterator iterator = set.iterator();
+    while (iterator.hasNext()) {
+      Map.Entry mapEntry = (Map.Entry) iterator.next();
+      if (mapEntry.getKey().equals(ParameterConstant.FIRST_NAME)) {
+        student.setFirstName((String) mapEntry.getValue());
+      }
+      if (mapEntry.getKey().equals(ParameterConstant.MIDDLE_NAME)) {
+        student.setMiddleName((String) mapEntry.getValue());
+      }
+      if (mapEntry.getKey().equals(ParameterConstant.LAST_NAME)) {
+        student.setLastName((String) mapEntry.getValue());
+      }
+      if (mapEntry.getKey().equals(ParameterConstant.ADDRESS)) {
+        student.setAddress((String) mapEntry.getValue());
+      }
+      if (mapEntry.getKey().equals(ParameterConstant.GRADE)) {
+        try{
+          student.setGrade(Integer.parseInt((String) mapEntry.getValue()));
+        } catch(NumberFormatException e){
+          error.put(AttributeConstant.ERROR_GRADE,AppConstant.TRUE);
+        }
+      }
+    }
+    return student;
+  }
+
+  public HashMap validate(Student student) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)